cql3/statements/select_statement.cc

```diff
@@ -439,9 +439,25 @@ generate_base_key_from_index_pk(const partition_key& index_pk, const std::option
 
     for (const column_definition& base_col : base_columns) {
         const column_definition* view_col = view_schema.view_info()->view_column(base_col);
+        if (!view_col) {
+            throw std::runtime_error(format("Base key column not found in the view: {}", base_col.name_as_text()));
+        }
+        if (base_col.type != view_col->type) {
+            throw std::runtime_error(format("Mismatched types for base and view columns {}: {} and {}",
+                    base_col.name_as_text(), base_col.type->cql3_type_name(), view_col->type->cql3_type_name()));
+        }
         if (view_col->is_partition_key()) {
             exploded_base_key.push_back(index_pk.get_component(view_schema, view_col->id));
-        } else if (index_ck) {
+        } else {
+            if (!view_col->is_clustering_key()) {
+                throw std::runtime_error(
+                        format("Base primary key column is not a primary key column in the materialized view: {} (kind: {})",
+                                view_col->name_as_text(), to_sstring(view_col->kind)));
+            }
+            if (!index_ck) {
+                throw std::runtime_error(format("Base primary key column is not present in the materialized view key: {}",
+                        view_col->name_as_text()));
```

Comment:
I can't make sense of this check, can you please verify and/or explain it to me? How does "index_ck" (what is it?) not existing related to the message you print, "base primary key column not present", and if it's base column, why do you print view_col)?

Reply:
I'll try to rephrase this message. What I wanted to express is that we reached the following situation:
* we want to get the value which corresponds to the base key column `base_col` - it should either be part of the index partition key or its clustering key
* we already determined that it's not part of the index partition key
* but, now we see that the paging state from the indexed query, which is supposed to store index's partition and clustering keys, does not have a clustering key after all! Which means that somehow the column that corresponds to the base column `base_col` was not shipped to us in the paging state, which must be a bug

To sum up, by `base column ${NAME}` I really mean `a column in the index, which corresponds to a base column with given name ${NAME}`. It's not easy to express that in a single small error message :)